specs/multispec.yml

```diff
@@ -0,0 +1,79 @@
+version: 1
+
+specs:
+  distroinfo:
+    centos:
+      distros:
+        - centos-7-x86_64
+      s2i_base: centos/s2i-base-centos7
+      org: "centos"
+      prod: "centos7"
+      logos: "centos-logos"
+      enablerepo: "--enablerepo=centosplus"
+      preinstall_cmd: ['yum install -y centos-release-scl']
+      python_pkgs: ['{{ spec.scl }}', '{{ spec.scl }}-python-devel',
+                    '{{ spec.scl }}-python-setuptools', '{{ spec.scl }}-python-pip']
+      base_pkgs: ['nss_wrapper', 'httpd24', 'httpd24-httpd-devel', 'httpd24-mod_ssl',
+                  'httpd24-mod_auth_kerb', 'httpd24-mod_ldap', 'httpd24-mod_session',
+                  'atlas-devel', 'gcc-gfortran', 'libffi-devel', 'libtool-ltdl enchant']
+      extra_pkgs: ['libjpeg-turbo', 'libjpeg-turbo-devel']
+
+    rhel7:
+      distros:
+        - rhel-7-x86_64
+      s2i_base: rhscl/s2i-base-rhel7
+      org: "rhscl"
+      prod: "rhel7"
+      logos: "redhat-logos"
+      component_prefix: "rh-"
+      preinstall_cmd: ['yum install -y yum-utils',
+                      'prepare-yum-repositories rhel-server-rhscl-7-rpms']
+      python_pkgs: ['{{ spec.scl }}', '{{ spec.scl }}-python-devel',
+                    '{{ spec.scl }}-python-setuptools', '{{ spec.scl }}-python-pip']
+      base_pkgs: ['nss_wrapper', 'httpd24', 'httpd24-httpd-devel', 'httpd24-mod_ssl',
+                  'httpd24-mod_auth_kerb', 'httpd24-mod_ldap', 'httpd24-mod_session',
+                  'atlas-devel', 'gcc-gfortran', 'libffi-devel', 'libtool-ltdl enchant']
+
+    fedora:
+      distros:
+        - fedora-27-x86_64
+      s2i_base: registry.fedoraproject.org/f27/s2i-base
+      img_tag: "latest"
+      python_pkgs: ['python3', 'python3-devel', 'python3-setuptools', 'python3-pip',
+                    'python3-virtualenv']
+      base_pkgs: ['nss_wrapper', 'httpd', 'httpd-devel', 'atlas-devel', 'gcc-gfortran',
+                  'libffi-devel', 'libtool-ltdl enchant']
+
+  version:
+    "3.6":
+      version: "3.6"
+      short_ver: "36"
+      img_version: "1"
```

Comment:
Could you move the "img_version" to the distro instead, ideally into a dictionary as suggested above, or as "python36_img_version" for better readability?

Reply:
Img_version is not distro specific. it is used on more places not only the image tag https://github.com/mcyprian/s2i-python-container/commit/cabe677976b28eaca14f26d952fce561a7715460#diff-c523d7243e565a463b4a2d07a9e2c40fR4. It was introduced in version 3.6 so it is logical to have it defined only for this version. After python 3.7 is added something like python36_img_version won't work.

Reply:
Dockerfile for centos simply doesn't contain tag in FROM declaration https://github.com/mcyprian/s2i-python-container/blob/ultimate-distgen/3.6/Dockerfile#L3